manta-pay/src/crypto/hash/poseidon_parameter_generation/constants.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2019-2022 Manta Network.
+// This file is part of manta-rs.
+//
+// manta-rs is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// manta-rs is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with manta-rs.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Generate constants for optimized poseidon hash
+
+use super::round_numbers::calc_round_numbers;
+use super::round_constants::generate_round_constants;
+use super::preprocessing::compress_round_constants;
+use super::matrix::Matrix;
+use super::mds::{factor_to_sparse_matrixes, MdsMatrices, SparseMatrix};
+use crate::crypto::hash::poseidon::Specification;
+use core::fmt::Debug;
+use alloc::vec::Vec;
+
+// TODO: shall we put constant generation code to compile time?

```

Comment:
Concluded in last meeting, we don't need to put constant generation in compile time (should be in manta-sdk?)

Reply:
@bhgomes what's your opinion on this issue?

Reply:
No need, just try to put the spec generation at compile time like the number of full and partial rounds and the alpha coefficient.